src/transaction/log_applier.c

```diff
@@ -7043,142 +7043,163 @@ la_print_log_arv_header (const char *database_name, LOG_ARV_HEADER * hdr, bool v
 }
 
 /*
- * la_log_page_check() - test the transaction log
- *   return: void
- *   log_path: log path
- *   page_num: test page number
+ * la_get_applyinfo_applied_log_info() - 
+ *   return: NO_ERROR or error code
+ *   database_name(in): db name
+ *   log_path(in): real path of log file
+ *   check_replica_info(in): whether replica node execute applyinfo utility
+ *   verbose(in): applyinfo -v option
+ *   applied_final_lsa(in): information for delayed applying copied log
+ *
  */
 int
-la_log_page_check (const char *database_name, const char *log_path, INT64 page_num, bool check_applied_info,
-		   bool check_copied_info, bool check_replica_info, bool verbose, LOG_LSA * copied_eof_lsa,
-		   LOG_LSA * copied_append_lsa, LOG_LSA * applied_final_lsa)
+la_get_applyinfo_applied_log_info (const char *database_name, const char *log_path, bool check_replica_info,
+				   bool verbose, LOG_LSA * applied_final_lsa)
 {
   int error = NO_ERROR;
   int res;
-  char *atchar;
-  char active_log_path[PATH_MAX];
   char *replica_time_bound_str;
 
   assert (database_name != NULL);
   assert (log_path != NULL);
 
-  atchar = (char *) strchr (database_name, '@');
-  if (atchar)
-    {
-      *atchar = '\0';
-    }
+  LA_HA_APPLY_INFO ha_apply_info;
+  char timebuf[1024];
 
-  /* init la_Info */
-  la_init (log_path, 0);
+  la_init_ha_apply_info (&ha_apply_info);
 
-  if (check_applied_info)
+  res = la_get_ha_apply_info (log_path, database_name, &ha_apply_info);
+  if ((res <= 0) || (ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0))
     {
-      LA_HA_APPLY_INFO ha_apply_info;
-      char timebuf[1024];
-
-      la_init_ha_apply_info (&ha_apply_info);
-
-      res = la_get_ha_apply_info (log_path, database_name, &ha_apply_info);
-      if ((res <= 0) || (ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0))
+      error = res;
+      if ((res < 0) && (ha_apply_info.creation_time.date != 0 || ha_apply_info.creation_time.time != 0))
```

Comment:
How about just `if (res < 0)`? `error` is set to something, so that we can print `db_error_string(3)` below. And could you tell me what `(ha_apply_info.creation_time.date == 0 && ha_apply_info.creation_time.time == 0)` means?

Reply:
`ha_apply_info.creation_time` looks like safe-guard that res <=0.
Because it is initialized `applyinfo` first and then stored data `la_get_ha_apply_info`. 

And if res is equal and less than 0, it means no chance to ha_apply_info.creation_time is set.